### PR TITLE
Revamp sequencer and aggregator integration with ethtxmanager

### LINKED ISSUE #1271: Better handling for sending L1 txs
Instead of sending txs synchronously from multiple points of the code (sequencer and aggregator), store the txs that need to be sent on a DB, and have an additional component that deals with sending those txs.

This should enable:

- Better security (only one component needs the private key)
- More resilient: components can go down, but the new component still keeps track of pending txs
- Better nonce management: on the DB we can store what nonce is used for each tx. If facing a mismatched nonce this could even be edited manually by the admins

### PR DESCRIPTION
**Pull request by tclemos**
Closes #1271 and #1262.

### Reviewers

### CODE REVIEW

File: ethtxmanager/ethtxmanager.go [hunk 1]
```diff
@@ -7,157 +7,283 @@ package ethtxmanager
...
-				nonce = new(big.Int).SetUint64(tx.Nonce())
-				gasPrice = increaseGasPrice(tx.GasPrice(), c.cfg.PercentageToIncreaseGasPrice)
-				log.Infof("tx %s reached timeout, retrying with gas price = %d", tx.Hash(), gasPrice)
-				continue
+			// if the tx is already know, refresh the update date to give it more time to get mined
+			if errors.Is(err, core.ErrAlreadyKnown) {
+				err := c.state.UpdateSequenceGroupTx(ctx, sequenceGroup.TxHash, sequenceGroup.TxHash, nil)
+				if err != nil {
+					log.Errorf("give it more time to the sequence group related to the batches %v to get mined: %v", sequenceGroup.BatchNumbers, err)
+				}
```
Comment (ARR552) [inline ethtxmanager/ethtxmanager.go:215]: This sounds weird for me. If this error happens, it means that even the gasPrice of the resent tx is the same. In this case, wouldn't it be better to increase the gasPrice even more??

File: sequencer/sequencer.go [hunk 2]
```diff
@@ -91,16 +90,26 @@ func (s *Sequencer) Start(ctx context.Context) {
...
 		}
 	}()
+	// send and monitor persisted sequences to L1
+	go func() {
+		for {
+			s.txManager.SyncPendingSequences()
+			waitTick(ctx, tickerSendSequence)
+		}
+	}()
+
```
Comment (ARR552) [inline sequencer/sequencer.go:112]: I thought the ethtxmanager should be autonomous

File: ethtxmanager/ethtxmanager.go [hunk 3]
```diff
@@ -7,157 +7,283 @@ package ethtxmanager
...
+		// recalculated with the current prices of the network
+		tx, err := c.ethMan.VerifyBatch(proof.BatchNumber, proof.Proof, 0, nonce)
+		if err != nil {
+			// if the tx is already know, refresh the update date to give it more time to get mined
+			if errors.Is(err, core.ErrAlreadyKnown) {
+				err := c.state.UpdateProofTx(ctx, proof.BatchNumber, *proof.TxHash, nil)
+				if err != nil {
+					log.Errorf("give it more time to the proof related to the batch %v to get mined: %v", proof.BatchNumber, err)
+				}
+				return
```
[the remark this answers is not in the document]
Comment (tclemos) [reply ethtxmanager/ethtxmanager.go:274]: I don't know, we can add this feature, but I think this is not needed for now, we just need to make sure the tx get mined, and following the network costs, we can achieve this even if the tx takes a bit to get mined. 
Also, this error can be forced if the interval is too short.

File: ethtxmanager/ethtxmanager.go (hunk 1 above)
Comment (tclemos) [reply ethtxmanager/ethtxmanager.go:215]: exaplained above

File: sequencer/sequencer.go (hunk 2 above)
Comment (tclemos) [reply sequencer/sequencer.go:112]: I've explained this here: https://github.com/0xPolygonHermez/zkevm-node/issues/1271#issuecomment-1288968113

**Review by ToniRamirezM — APPROVED** [review]

File: sequencer/closingchecker.go [hunk 4]
```diff
@@ -66,23 +66,15 @@ func (s *Sequencer) shouldCloseDueToNewDeposits(ctx context.Context) (bool, erro
...
-	}
-	isPreviousBatchVirtualized, err := s.state.IsBatchVirtualized(ctx, lastBatchNumber-1, nil)
-	if err != nil {
-		log.Errorf("failed to get last virtual batch num, err: %w", err)
-		return false, err
-	}
-	if time.Unix(s.sequenceInProgress.Timestamp, 0).Add(s.cfg.MaxTimeForBatchToBeOpen.Duration).Before(time.Now()) &&
-		isPreviousBatchVirtualized && len(s.sequenceInProgress.Txs) > 0 {
+func (s *Sequencer) shouldCloseTooLongSinceLastSequence(ctx context.Context) bool {
+	timeoutExpired := s.sequenceInProgress.Timestamp.Add(s.cfg.MaxTimeForBatchToBeOpen.Duration).Before(time.Now())
```
Comment (Mikelle) [inline sequencer/closingchecker.go:72]: This check is not working as expected. The batch could be open any time long, but the time from the previous virtualized batch shouldn't be more, than a time.Now()-value from the config.
https://github.com/0xPolygonHermez/zkevm-node/issues/1245

Comment (tclemos) [reply sequencer/closingchecker.go:72]: the sequencer logic was changed to not depend on virtualized batches, now it just depends on the sequences created in the DB.

**Pull request merged**